packetbeat/protos/mysql/mysql.go

```diff
@@ -667,8 +785,161 @@ func (mysql *mysqlPlugin) receivedMysqlResponse(msg *mysqlMessage) {
 	mysql.publishTransaction(trans)
 	mysql.transactions.Delete(trans.tuple.Hashable())
 
-	logp.Debug("mysql", "Mysql transaction completed: %s", trans.mysql)
-	logp.Debug("mysql", "%s", trans.responseRaw)
+	logp.Debug("mysql", "Mysql transaction completed: %s %s %s", trans.query, trans.param, trans.mysql)
+	//	logp.Debug("mysql", "%s", trans.responseRaw)
+}
+
+func (mysql *mysqlPlugin) parseMysqlExecuteStatement(data []byte, stmtdata *mysqlStmtData) []string {
+
+	var paramType, paramUnsigned uint8
+	nparamType := []uint8{}
+	paramString := []string{}
+	nparam := stmtdata.numOfParameters
+	offset := 0
+	// mysql hdr
+	offset += 4
+	// cmd type
+	offset++
+	// stmt id
+	offset += 4
+	// flags
+	offset++
+	// iterations
+	offset += 4
+	// null-bitmap
+	if nparam > 0 {
+		offset += (nparam + 7) / 8
+	}
+	// stmt bound
+	stmtBound := data[offset]
```

Comment:
Is `data` guaranteed to have at least `offset` bytes here? If not, a check should be made before accessing to avoid "index out of bound" panics.

Reply:
You are right, a check should be made before accessing the data because some illegal packet will conflict with the protocol specification